### PR TITLE
Api800/server hardware type

### PR DESCRIPTION
**Pull request by AnushaY1916**
### Description
API 800 Support for Server hardware Type

### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for go 1.11 + gofmt checks .
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/server_hardware_type.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"os"
+)
+
+func main() {
+	var (
+		clientOV                  *ov.OVClient
+		server_hardware_type_name = "BL460c G7 1"
+		uri                       = "/rest/server-hardware-types/" + "70BDABAA-87D1-4A39-9270-047D08B5C447"
+		sort                      = "name:desc"
+		//filter = "created equals '2018-01-24T16:24:14.330Z'"
```
Comment (soodpr) [inline examples/server_hardware_type.go]: remove unnecessary code

File: examples/server_hardware_type.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/HewlettPackard/oneview-golang/ov"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"os"
+)
+
+func main() {
+	var (
+		clientOV                  *ov.OVClient
+		server_hardware_type_name = "BL460c G7 1"
+		uri                       = "/rest/server-hardware-types/" + "70BDABAA-87D1-4A39-9270-047D08B5C447"
```
Comment (soodpr) [inline examples/server_hardware_type.go]: remove concatenation of the string

File: endpoints-support.md [hunk 3]
```diff
@@ -99,8 +99,10 @@
...
 |     **Server Hardware Types**                                                                                                                     |
-|<sub>/rest/server-hardware-types</sub>                                                   | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |
-|<sub>/rest/server-hardware-types/{id}</sub>                                              | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   |
+|<sub>/rest/server-hardware-types</sub>                                                   | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/server-hardware-types/{id}</sub>                                              | GET      | :white_check_mark:   | :white_check_mark:   | :heavy_multiplication_x:   | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/server-hardware-types/{id}</sub>                                              | PUT      | :heavy_minus_mark:   | :heavy_minus_mark:   | :heavy_minus_sign:         | :heavy_multiplication_x:   | :heavy_multiplication_x:   |
```
Comment (soodpr) [inline endpoints-support.md]: make this file aligned properly

File: examples/server_hardware_type.go (hunk 1 above)
Comment (AnushaY1916) [reply examples/server_hardware_type.go]: Removed

File: examples/server_hardware_type.go (hunk 2 above)
Comment (AnushaY1916) [reply examples/server_hardware_type.go]: Removed concatenation

File: endpoints-support.md (hunk 3 above)
Comment (AnushaY1916) [reply endpoints-support.md]: Aligned properly

**Pull request merged**